File: Daily/sbgo/Leetcode524-通过删除匹配到最长单词.go
package sbgo

func findLongestWord(s string, dictionary []string) string {
	ans := ""
	for _, x := range dictionary {
		i := 0
		for j := range s {
			if s[j] == x[i] {
				i++
				if i == len(x) {
					if len(x) > len(ans) || len(x) == len(ans) && x < ans {
						ans = x
					}
					break
				}
			}
		}
	}
	return ans
}

File: Daily/PY/Leetcode524-通过删除匹配到最长单词.py
from typing import List
class Solution:
    def findLongestWord(self, s: str, dictionary: List[str]) -> str:
        res = ''
        dictionary.sort()
        for x in dictionary:
            i,j = 0,0
            while i < len(s) and j < len(x):
                if s[i] == x[j]:
                    i += 1
                    j += 1
                else:
                    i += 1
            if j == len(x) and j > len(res):
                res = x
        return res